### DISCUSSION TITLE
Can you set an offset when getting messages?

### DISCUSSION THREAD

**Discussion by pikachub2005** [discordjs/discord.js#7426] [Q&A] [answered]
In `await channel.messages.fetch({limit:20})` can you add an offset? Or should I just get the last 25 and delete the first 5

**Comment by kyranet** [comment, marked as answer]
You can use `before` and `after` to paginate thru the messages, see the [documentation](https://discord.js.org/#/docs/discord.js/stable/typedef/ChannelLogsQueryOptions).

**Reply by pikachub2005** [reply]
ty!

**Comment by ImRodry** [comment]
If you know the id of the last message you don’t want you can send that as “after”: https://discord.js.org/#/docs/discord.js/stable/typedef/ChannelLogsQueryOptions
Otherwise I recommend getting 25 and then getting the last 20
